lms/djangoapps/teams/views.py

```diff
@@ -71,11 +71,12 @@ def get(self, request, course_id):
         course_key = CourseKey.from_string(course_id)
         course = get_course_with_access(request.user, "load", course_key)
 
+        is_staff = bool(has_access(request.user, 'staff', course, course.id))
+
         if not is_feature_enabled(course):
             raise Http404
 
-        if not CourseEnrollment.is_enrolled(request.user, course.id) and \
-                not has_access(request.user, 'staff', course, course.id):
+        if not CourseEnrollment.is_enrolled(request.user, course.id) and not is_staff:
```

Comment:
Nit: I would leave the code as it was so that we don't evaluate the has_access call unless the caller is not enrolled in the course.